pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -874,6 +937,65 @@ func (pl *DynamicResources) bindClaim(ctx context.Context, state *stateData, ind
 	return claim, nil
 }
 
+// isClaimBound checks whether a given resource claim is successfully
+// bound to a device.
+func (pl *DynamicResources) isClaimBound(claim *resourceapi.ResourceClaim) (bool, error) {
+	for _, deviceRequest := range claim.Status.Allocation.Devices.Results {
+		if len(deviceRequest.BindingConditions) == 0 {
+			continue
+		}
+		deviceStatus := getAllocatedDeviceStatus(claim, &deviceRequest)
+		if deviceStatus == nil {
+			return false, nil
+		}
+		for _, cond := range deviceRequest.BindingFailureConditions {
+			if apimeta.IsStatusConditionTrue(deviceStatus.Conditions, cond) {
+				return false, fmt.Errorf("claim %s failed to bind", claim.Name)
+			}
+		}
+		for _, cond := range deviceRequest.BindingConditions {
+			if !apimeta.IsStatusConditionTrue(deviceStatus.Conditions, cond) {
+				return false, nil
+			}
+		}
+	}
+	return true, nil
+}
+
+// hasDeviceBindingStatus checks the binding status of devices within the
+// given state claims.
+func (pl *DynamicResources) hasDeviceBindingStatus(ctx context.Context, state *stateData, pod *v1.Pod, nodeName string) (bool, error) {
+	for claimIndex, claim := range state.claims {
+		claim, err := pl.clientset.ResourceV1beta1().ResourceClaims(claim.Namespace).Get(ctx, claim.Name, metav1.GetOptions{})
```

Comment:
Shouldn't this check the assume cache instead of doing a GET call?

Reply:
I think that a GET is required because I need to receive updates by the outer DRA driver. Will "assume cache" allow for that?

Reply:
Yes. The cache gets updated by watching changes on the apiserver.

Reply:
OK, I will change it,